dashboard/app/models/school_stats_by_year.rb

```diff
@@ -50,20 +50,48 @@ class SchoolStatsByYear < ApplicationRecord
   # Requires a block to parse the row.
   # @param filename [String] The CSV file name.
   # @param options [Hash] Optional, the CSV file parsing options.
-  def self.merge_from_csv(filename, options = {col_sep: "\t", headers: true, quote_char: "\x00"})
-    CSV.read(filename, options).each do |row|
-      parsed = yield row
-      loaded = find_by(primary_keys.map(&:to_sym).map {|k| [k, parsed[k]]}.to_h)
-      if loaded.nil?
-        begin
-          SchoolStatsByYear.new(parsed).save!
-        rescue ActiveRecord::InvalidForeignKey
-          puts parsed[:school_id]
+  # @param dry_run [Boolean] Optional, roll back any db transactions after processing.
+  def self.merge_from_csv(filename, options = {col_sep: "\t", headers: true, quote_char: "\x00"}, dry_run = false)
+    school_stats = []
+    new_schools = 0
+    updated_schools = 0
+    unchanged_schools = 0
+    errored_schools = []
+
+    ActiveRecord::Base.transaction do
+      school_stats = CSV.read(filename, options).each do |row|
```

Comment:
Do you need this variable?

Reply:
nope! left over from some previous logging